flycheck.el

```diff
@@ -7772,6 +7772,12 @@ dirname is test or else default."
                    (directory-file-name
                     (file-name-directory buffer-file-name)))))
     "test")
+   ((and buffer-file-name
```

Comment:
Instead of duplicating the second condition, we merge conditions 2 and 3 as:
```elisp
((seq-contains '("test" "eqc")
                 (and buffer-file-name
                      (file-name-base
                       (directory-file-name
                        (file-name-directory buffer-file-name)))))
```
Taking advantage of `cond` returning the value of `CONDITION` when `BODY` is empty.  We can also simplify the first condition as:
```elisp
(flycheck-erlang-rebar3-profile)
```
along the same lines.

Reply:
@fmdkdd - thanks for the suggestions, here you go